ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Log.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.math;
+
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+
+@UdfDescription(
+    name = "log",
```

Comment:
Maybe we should name this function `ln` and use `log` for a function that accepts a base? I don't really have a strong preference though - different systems seem to do different things here. Postgres uses ln. Mysql uses both ln and log, and log w/ 1 parameter takes a natural log logarithm - so we're consistent with that.